plugins/inputs/t128_metrics/t128_metrics_test.go

```diff
@@ -0,0 +1,512 @@
+		ConfiguredMetrics: []plugin.ConfiguredMetric{{
+			"test-metric",
+			map[string]string{"test-field": "stats/test"},
+			map[string][]string{"interface": {}},
+		}},
+		Endpoints: []Endpoint{{"/stats/test", 200, `{"parameters": [{"name": "interface", "itemize": true}]}`, "[]"}},
+	},
+	{
+		Name: "includes parameter filter values",
+		ConfiguredMetrics: []plugin.ConfiguredMetric{{
+			"test-metric",
+			map[string]string{"test-field": "stats/test"},
+			map[string][]string{"interface": {"intf1", "intf2"}},
+		}},
+		Endpoints: []Endpoint{{"/stats/test", 200, `{
+				"parameters": [
+					{"name": "interface", "values": ["intf1", "intf2"], "itemize": true}
+				]
+			}`,
+			"[]"}},
+	},
+	{
+		Name: "includes multiple parameter filters",
+		ConfiguredMetrics: []plugin.ConfiguredMetric{{
+			"test-metric",
+			map[string]string{"test-field": "stats/test"},
+			map[string][]string{
+				"interface": {"intf1", "intf2"},
+				"node":      {"node1", "node2"},
+				"other":     {}},
+		}},
+		Endpoints: []Endpoint{{"/stats/test", 200, `{
+				"parameters": [
+					{"name": "interface", "values": ["intf1", "intf2"], "itemize": true},
+					{"name": "node", "values": ["node1", "node2"], "itemize": true},
+					{"name": "other", "itemize": true}
+				]
+			}`,
+			"[]"}},
+	},
+}
+
+func TestT128MetricsResponseProcessing(t *testing.T) {
+	for _, testCase := range ResponseProcessingTestCases {
+		t.Run(testCase.Name, func(t *testing.T) {
+			fakeServer := createTestServer(t, testCase.Endpoints)
+			defer fakeServer.Close()
+
+			plugin := &plugin.T128Metrics{
+				BaseURL:                 fakeServer.URL,
+				MaxSimultaneousRequests: 20,
+				ConfiguredMetrics:       testCase.ConfiguredMetrics,
+			}
+
+			var acc testutil.Accumulator
+			plugin.Init()
+
+			plugin.Gather(&acc)
+
+			var errorStrings []string = nil
+			for _, err := range acc.Errors {
+				errorStrings = append(errorStrings, err.Error())
+			}
+
+			require.ElementsMatch(t, testCase.ExpectedErrors, errorStrings)
+
+			// Timestamps aren't important, but need to match
+			for _, m := range acc.Metrics {
+				m.Time = time.Time{}
+			}
+
+			// Avoid specifying this unused type for each field
+			for _, m := range testCase.ExpectedMetrics {
+				m.Type = telegraf.Untyped
+			}
+
+			require.ElementsMatch(t, testCase.ExpectedMetrics, acc.Metrics)
+		})
+	}
+}
+
+func TestT128MetricsRequestFormation(t *testing.T) {
+	for _, testCase := range RequestFormationTestCases {
+		t.Run(testCase.Name, func(t *testing.T) {
+			fakeServer := createTestServer(t, testCase.Endpoints)
+			defer fakeServer.Close()
+
+			plugin := &plugin.T128Metrics{
+				BaseURL:                 fakeServer.URL,
+				MaxSimultaneousRequests: 20,
+				ConfiguredMetrics:       testCase.ConfiguredMetrics,
+			}
+
+			var acc testutil.Accumulator
+			plugin.Init()
+
+			require.NoError(t, acc.GatherError(plugin.Gather))
+		})
+	}
+}
+
+func TestT128MetricsErrorResponses(t *testing.T) {
```

Comment:
I'm just now seeing this duplicate test. I ended up moving the error handling test cases into the response processing test. This one should be removed.